tests/integration/high_availability/test_replication.py

```diff
@@ -53,6 +56,50 @@ async def test_exporter_endpoints(ops_test: OpsTest, mysql_charm_series: str) ->
     http = urllib3.PoolManager()
 
     for unit in application.units:
+        _, output, _ = await ops_test.juju(
+            "ssh", unit.name, "sudo", "snap", "services", "charmed-mysql.mysqld-exporter"
+        )
+        assert output.split("\n")[1].split()[2] == "inactive"
+
+        return_code, _, _ = await ops_test.juju(
+            "ssh", unit.name, "sudo", "snap", "set", "charmed-mysql", "exporter.user=monitoring"
+        )
+        assert return_code == 0
+
+        monitoring_credentials = await fetch_credentials(unit, "monitoring")
+        return_code, _, _ = await ops_test.juju(
+            "ssh",
+            unit.name,
+            "sudo",
+            "snap",
+            "set",
+            "charmed-mysql",
+            f"exporter.password={monitoring_credentials['password']}",
+        )
+        assert return_code == 0
+
+        return_code, _, _ = await ops_test.juju(
+            "ssh", unit.name, "sudo", "snap", "start", "charmed-mysql.mysqld-exporter"
+        )
+        assert return_code == 0
```

Comment:
because we are not only starting the exporter service when a relation with cos is formed. by default, the service is disabled. it would be too much to set up a cos bundle to test the exporter, so this is a lighter version to make sure that the exporter service still works

Reply:
IMO I don't think it's that much of a hassle, since you won't need to setup a full cos deployment, but just a relation with the subordinated `grafana-agent` charm, which is fairly light weight